### ISSUE TITLE
Deleting nodes doesn't change labels

### ISSUE THREAD

**Issue by filchristou** [JuliaGraphs/MetaGraphsNext.jl#33]
When I delete a node of `MetaGraph`, then the indices are not updated and errors happen.

```julia
# From the docs example
julia> colors = MetaGraph( Graph(), VertexData = String, EdgeData = Symbol, graph_data = "graph_of_colors")
Meta graph based on a {0, 0} undirected simple Int64 graph with vertex labels of type Symbol, vertex metadata of type String, edge metadata of type Symbol, graph metadata given by "graph_of_colors", and default weight 1.0
julia> colors[:red] = "warm";
julia> colors[:yellow] = "warm";
julia> colors[:blue] = "warm";  # wrong metadata
julia> colors[:blue] = "cool";
```
I would say that the following code should always return the label
```julia
julia> label_for(colors, code_for(colors, :blue))
:blue
```
but if I delete a vertex, then it fails:
```julia
julia> rem_vertex!(colors, 1)
julia> label_for(colors, code_for(colors, :blue))
ERROR: KeyError: key 3 not found
Stacktrace:
 [1] getindex
   @ ./dict.jl:481 [inlined]
 [2] label_for(g::MetaUndirectedGraph{Int64, Symbol, SimpleGraph{Int64}, String, Symbol, String, MetaGraphsNext.var"#3#5", Float64}, v::Int64)
   @ MetaGraphsNext ~/.julia/packages/MetaGraphsNext/BYnZO/src/graphs.jl:36
 [3] top-level scope
   @ REPL[30]:1
```

I think the reason is the vertex_properties are not updated:
```julia
julia> colors.vertex_labels
Dict{Int64, Symbol} with 2 entries:
  2 => :yellow
  1 => :blue

julia> colors.vertex_properties
Dict{Symbol, Tuple{Int64, String}} with 2 entries:
  :yellow => (2, "warm")
  :blue   => (3, "cool")
```

**Comment by bramtayl** [comment]
Thanks, that's definitely a bug

**Comment by bramtayl** [comment]
Closed by #33
